### ISSUE TITLE
appcenter-cli v2.6.0 causes internal server error on upload artefact

### ISSUE THREAD

**Issue by jpelgrim** [wzieba/AppCenter-Github-Action#13]
Version 2.6.0 of the `appcenter-cli` package is causing internal server errors when attempting to upload the artefact where v2.5.4 is (still) running fine. Maybe we should pin the `appcenter-cli` version to v2.5.4 and await if e.g. v2.6.1 fixes this issue?

**Comment by wzieba** [comment]
Sure, thank you for reporting. I'll allow myself to use commit from your fork.

**Comment by wzieba** [comment]
@jpelgrim I cannot reproduce your issue, I've just re-run tests from `master` branch: https://github.com/wzieba/AppCenter-Github-Action/runs/795362520

This installed version `2.6.0` and works fine. Could you please share some sample workflow file or logs?

**Comment by jpelgrim** [comment]
Removed all hashes / ids, etc. Hope the log is still useful. It probably boils down to the `Internal server error. Correlation ID: correlation-id-removed` on the last line.

Again, this might just be an appcenter-cli issue of course. I don't know. Nothing changed and our builds started to fail today.

```
#######################
### SUCCESFUL BUILD ###
#######################

2020-06-19T15:12:07.9625733Z ##[command]/usr/bin/docker run --name name-removed --label label-removed --workdir /github/workspace --rm -e JAVA_HOME -e JAVA_HOME_12.0.2_x64 -e FLUTTER_HOME -e INPUT_APPNAME -e INPUT_TOKEN -e INPUT_GROUP -e INPUT_FILE -e INPUT_RELEASENOTES -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORYEPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/project-name-removed/project-name-removed":"/github/workspace" label-removed:name-removed  "group-name-removed/app-name-removed" "***" "Team Name Removed" "build/app/outputs/apk/release/app-name-removed.apk" "Merge pull request #173 from OrganizationNameRemoved/feature/branch-name-removed  Commit message removed "
2020-06-19T15:12:13.8080917Z Using appcenter-cli version: 2.5.4
2020-06-19T15:12:13.8165692Z logFilter, request: {
2020-06-19T15:12:13.8165826Z   "rawResponse": false,
2020-06-19T15:12:13.8165959Z   "queryString": {},
2020-06-19T15:12:13.8166518Z   "url": "https://api.appcenter.ms/v0.1/apps/group-name-removed/app-name-removed/distribution_groups/Team%20Name%20Removed/members",
2020-06-19T15:12:13.8166665Z   "method": "GET",
2020-06-19T15:12:13.8166767Z   "headers": {
2020-06-19T15:12:13.8167004Z     "Content-Type": "application/json; charset=utf-8",
2020-06-19T15:12:13.8167267Z     "user-agent": "appcenterCli/2.5.4 NodeJS/v12.10.0 linux/5.3.0-1028-azure",
2020-06-19T15:12:13.8167897Z     "x-api-token": "***"
2020-06-19T15:12:13.8168007Z   },
2020-06-19T15:12:13.8168106Z   "body": null
2020-06-19T15:12:13.8168213Z }
2020-06-19T15:12:14.1524907Z Response status code: 200
2020-06-19T15:12:14.1532461Z Body: [list of people removed]
2020-06-19T15:12:14.1558805Z logFilter, request: {
2020-06-19T15:12:14.1559101Z   "rawResponse": false,
2020-06-19T15:12:14.1559360Z   "queryString": {},
2020-06-19T15:12:14.1560466Z   "url": "https://api.appcenter.ms/v0.1/apps/group-name-removed/app-name-removed/release_uploads",
2020-06-19T15:12:14.1560815Z   "method": "POST",
2020-06-19T15:12:14.1561064Z   "headers": {
2020-06-19T15:12:14.1561649Z     "Content-Type": "application/json; charset=utf-8",
2020-06-19T15:12:14.1562290Z     "user-agent": "appcenterCli/2.5.4 NodeJS/v12.10.0 linux/5.3.0-1028-azure",
2020-06-19T15:12:14.1563511Z     "x-api-token": "***"
2020-06-19T15:12:14.1563776Z   },
2020-06-19T15:12:14.1564018Z   "body": null
2020-06-19T15:12:14.1564255Z }
2020-06-19T15:12:14.4493915Z Response status code: 201
2020-06-19T15:12:14.4495820Z Body: {"upload_id":"upload-id-removed","upload_url":"https://upload.appcenter.ms/v0.1/apps/app-id-removed/uploads/upload-id-removed%7Capp-id-removed%7Casset-id-removed%7Chash-removed%3D%7Cfile.appcenter.ms","asset_id":"asset-id-removed","asset_domain":"https://file.appcenter.ms","asset_token":"?sv=2019-02-02&sr=c&si=asset-id-removed&sig=sig-removed&se=2020-06-24T15%3A17%3A14Z&t=distribution"}
2020-06-19T15:12:16.6849453Z logFilter, request: {
2020-06-19T15:12:16.6850100Z   "rawResponse": false,
2020-06-19T15:12:16.6850354Z   "queryString": {},
2020-06-19T15:12:16.6851802Z   "url": "https://api.appcenter.ms/v0.1/apps/group-name-removed/app-name-removed/release_uploads/upload-id-removed",
2020-06-19T15:12:16.6852180Z   "method": "PATCH",
2020-06-19T15:12:16.6855129Z   "headers": {
2020-06-19T15:12:16.6855996Z     "Content-Type": "application/json; charset=utf-8",
2020-06-19T15:12:16.6856669Z     "user-agent": "appcenterCli/2.5.4 NodeJS/v12.10.0 linux/5.3.0-1028-azure",
2020-06-19T15:12:16.6857911Z     "x-api-token": "***"
2020-06-19T15:12:16.6858437Z   },
2020-06-19T15:12:16.6858699Z   "body": "{\"status\":\"committed\"}"
2020-06-19T15:12:16.6858952Z }
2020-06-19T15:12:16.7908339Z Response status code: 200
2020-06-19T15:12:16.7910049Z Body: {"release_id":"103","release_url":"v0.1/apps/group-name-removed/app-name-removed/releases/103"}
2020-06-19T15:12:16.7944145Z logFilter, request: {
2020-06-19T15:12:16.7944689Z   "rawResponse": false,
2020-06-19T15:12:16.7944996Z   "queryString": {},
2020-06-19T15:12:16.7945751Z   "url": "https://api.appcenter.ms/v0.1/apps/group-name-removed/app-name-removed/releases/103",
2020-06-19T15:12:16.7946083Z   "method": "PUT",
2020-06-19T15:12:16.7946349Z   "headers": {
2020-06-19T15:12:16.7946805Z     "Content-Type": "application/json; charset=utf-8",
2020-06-19T15:12:16.7947321Z     "user-agent": "appcenterCli/2.5.4 NodeJS/v12.10.0 linux/5.3.0-1028-azure",
2020-06-19T15:12:16.7948305Z     "x-api-token": "***"
2020-06-19T15:12:16.7948602Z   },
2020-06-19T15:12:16.7949165Z   "body": "{\"release_notes\":\"Merge pull request #173 from OrganizationNameRemoved/feature/branch-name-removed  Commit message removed \"}"
2020-06-19T15:12:16.7949473Z }
2020-06-19T15:12:16.9049159Z Response status code: 200
2020-06-19T15:12:16.9051105Z Body: {"release_notes":"Merge pull request #173 from OrganizationNameRemoved/feature/branch-name-removed  Commit message removed "}
2020-06-19T15:12:16.9064725Z logFilter, request: {
2020-06-19T15:12:16.9065397Z   "rawResponse": false,
2020-06-19T15:12:16.9066377Z   "queryString": {},
2020-06-19T15:12:16.9067892Z   "url": "https://api.appcenter.ms/v0.1/apps/group-name-removed/app-name-removed/distribution_groups/Team%20Name%20Removed",
2020-06-19T15:12:16.9068492Z   "method": "GET",
2020-06-19T15:12:16.9068937Z   "headers": {
2020-06-19T15:12:16.9069749Z     "Content-Type": "application/json; charset=utf-8",
2020-06-19T15:12:16.9072664Z     "user-agent": "appcenterCli/2.5.4 NodeJS/v12.10.0 linux/5.3.0-1028-azure",
2020-06-19T15:12:16.9074492Z     "x-api-token": "***"
2020-06-19T15:12:16.9075029Z   },
2020-06-19T15:12:16.9075473Z   "body": null
2020-06-19T15:12:16.9076048Z }
2020-06-19T15:12:17.1205776Z Response status code: 200
2020-06-19T15:12:17.1206712Z Body: {"id":"id-removed","name":"Name Removed","origin":"appcenter","display_name":"Display Name Removed","is_public":false}
2020-06-19T15:12:17.1236741Z logFilter, request: {
2020-06-19T15:12:17.1236926Z   "rawResponse": false,
2020-06-19T15:12:17.1237063Z   "queryString": {},
2020-06-19T15:12:17.1237683Z   "url": "https://api.appcenter.ms/v0.1/apps/group-name-removed/app-name-removed/releases/103/groups",
2020-06-19T15:12:17.1237870Z   "method": "POST",
2020-06-19T15:12:17.1238004Z   "headers": {
2020-06-19T15:12:17.1238311Z     "Content-Type": "application/json; charset=utf-8",
2020-06-19T15:12:17.1238648Z     "user-agent": "appcenterCli/2.5.4 NodeJS/v12.10.0 linux/5.3.0-1028-azure",
2020-06-19T15:12:17.1260473Z     "x-api-token": "***"
2020-06-19T15:12:17.1260728Z   },
2020-06-19T15:12:17.1261291Z   "body": "{\"id\":\"id-re-mo-ved\",\"mandatory_update\":false,\"notify_testers\":true}"
2020-06-19T15:12:17.1261456Z }
2020-06-19T15:12:17.6140458Z Response status code: 201
2020-06-19T15:12:17.6141578Z Body: {"id":"id-removed","mandatory_update":false,"provisioning_status_url":null}
2020-06-19T15:12:17.6159898Z logFilter, request: {
2020-06-19T15:12:17.6160385Z   "rawResponse": false,
2020-06-19T15:12:17.6160594Z   "queryString": {},
2020-06-19T15:12:17.6161483Z   "url": "https://api.appcenter.ms/v0.1/apps/group-name-removed/app-name-removed/releases/103",
2020-06-19T15:12:17.6161759Z   "method": "GET",
2020-06-19T15:12:17.6161957Z   "headers": {
2020-06-19T15:12:17.6162413Z     "Content-Type": "application/json; charset=utf-8",
2020-06-19T15:12:17.6162918Z     "user-agent": "appcenterCli/2.5.4 NodeJS/v12.10.0 linux/5.3.0-1028-azure",
2020-06-19T15:12:17.6163923Z     "x-api-token": "***"
2020-06-19T15:12:17.6164278Z   },
2020-06-19T15:12:17.6164472Z   "body": null
2020-06-19T15:12:17.6164661Z }
2020-06-19T15:12:17.9017918Z Response status code: 200
2020-06-19T15:12:17.9022951Z Body: {"app_name":"app-name-removed","app_display_name":"App Display Name Removed","app_os":"Android","app_icon_url":"https://app.icon/url/removed","is_external_build":false,"origin":"appcenter","id":103,"version":"134","short_version":"0.6.5","size":32636540,"min_os":"5.0","android_min_api_level":"21","device_family":null,"bundle_identifier":"bundle.identifier.removed","fingerprint":"fingerprint-removed","uploaded_at":"2020-06-19T15:12:16.591Z","download_url":"https://downlo.ad/url/removed","enabled":true,"release_notes":"Merge pull request #173 from OrganizationNameRemoved/feature/branch-name-removed  Commit message removed ","package_hashes":["package-hash-removed"],"destinations":[{"destination_type":"group","id":"id-removed","name":"Name Removed"}],"destination_type":"group","distribution_groups":[{"id":"id-removed","name":"Name Removed"}]}
2020-06-19T15:12:17.9028530Z Release 0.6.5 (134) was successfully released to 11 testers in Team Name Removed

#####################
### FAILING BUILD ###
#####################

2020-06-22T10:14:45.5862289Z ##[command]/usr/bin/docker run --name name-removed --label label-removed --workdir /github/workspace --rm -e JAVA_HOME -e JAVA_HOME_12.0.2_x64 -e FLUTTER_HOME -e INPUT_APPNAME -e INPUT_TOKEN -e INPUT_GROUP -e INPUT_FILE -e INPUT_RELEASENOTES -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/project-name-removed/project-name-removed":"/github/workspace" label-removed:name-removed  "group-name-removed/app-name-removed" "***" "Team Name Removed" "build/app/outputs/apk/release/app-name-removed.apk" "Commit message removed"
2020-06-22T10:14:52.6463763Z Using appcenter-cli version: 2.6.0
2020-06-22T10:14:52.6538781Z logFilter, request: {
2020-06-22T10:14:52.6538946Z   "rawResponse": false,
2020-06-22T10:14:52.6539088Z   "queryString": {},
2020-06-22T10:14:52.6539762Z   "url": "https://api.appcenter.ms/v0.1/apps/group-name-removed/app-name-removed/distribution_groups/Team%20Name%20Removed/members",
2020-06-22T10:14:52.6539957Z   "method": "GET",
2020-06-22T10:14:52.6540097Z   "headers": {
2020-06-22T10:14:52.6540939Z     "Content-Type": "application/json; charset=utf-8",
2020-06-22T10:14:52.6541302Z     "user-agent": "appcenterCli/2.6.0 NodeJS/v12.10.0 linux/5.3.0-1028-azure",
2020-06-22T10:14:52.6542088Z     "x-api-token": "***"
2020-06-22T10:14:52.6542241Z   },
2020-06-22T10:14:52.6542593Z   "body": null
2020-06-22T10:14:52.6542732Z }
2020-06-22T10:14:53.0064394Z Response status code: 200
2020-06-22T10:14:53.0068914Z Body: [list of people removed]
2020-06-22T10:14:53.1360530Z Error: failed to create release upload for build/app/outputs/apk/release/app-name-removed.apk. Internal server error. Correlation ID: correlation-id-removed
```

**Comment by rvp-diconium** [comment]
+1 for this.
Exactly same issue, exactly same logs for success/failed builds.

**Comment by jpelgrim** [comment]
@wzieba isn't it an idea to pin e.g. v1.0.1 of the `AppCenter-GitHub-Action` action to v2.5.4 of [`appcenter-cli`](https://github.com/microsoft/appcenter-cli). Then, when a new `appcenter-cli` version is released e.g. create a v1.0.2 of the `AppCenter-GitHub-Action` action and pin that to e.g 2.6.0 in this case. That way people can easily switch back to v1.0.1 of this action if v1.0.2 is failing. I know, this means more maintenance for this project, but at least we know that our builds work with the same `appcenter-cli` version underneath.

**Comment by wzieba** [comment]
Thank you @jpelgrim for preparing log! I still can't understand why I can't reproduce this but as this will help you and other users of the lib, sure thing I'll pin the version.

**Comment by jpelgrim** [comment]
+ maybe add a new v1.0.1 version tag for this @wzieba?

**Comment by wzieba** [comment]
@jpelgrim sure, thank you for reminding. I've released it under `v1.1.1`, together with feature introduced in #4
